### PR TITLE
Implement historical trades call

### PR DESCRIPTION
**Pull request by paulofabiano**
This PR aims to support the Binance Spot endpoint calls for historical trades.
https://binance-docs.github.io/apidocs/spot/en/#old-trade-lookup

### CODE REVIEW

File: lib/binance.ex
Comment (Cinderella-Man) [inline lib/binance.ex:75]: would it be possible to compress those two functions into one with optional `fromId` handled in a similar fashion as in other places in the Binance module:

```
    arguments =
      %{
        symbol: symbol,
        limit: limit
      }
      |> Map.merge(
        unless(
          is_nil(from_id),
          do: %{fromId: from_id},
          else: %{}
        )
      )
```

Also, could you please use snake_case style for variable names?

File: lib/binance/rest/http_client.ex
Comment (Cinderella-Man) [inline lib/binance/rest/http_client.ex:100]: I think you should *not* implement a new function here but use the `unsigned_request_binance` function instead.

Docs above it need to be fixed as it will now accept get requests as well.

Then another clause to `defp do_unsigned_request(url, params, :get, headers) do` could be added where you would do the conversion to URL params.

**Review by Cinderella-Man — CHANGES_REQUESTED** [review]
Hello :wave: Thank you for your contribution but this requires a little bit of rethinking - I think it could be achieved in an easier way - I described it in my comments. Please let me know is everything clear.

File: lib/binance.ex
Comment (paulofabiano) [reply lib/binance.ex]: DONE

File: lib/binance/rest/http_client.ex
Comment (paulofabiano) [reply lib/binance/rest/http_client.ex]: DONE

**Comment by dvcrn** [comment]
@Cinderella-Man thanks a lot for the handling of this! Your input to this project has been very helpful. I was wondering if you'd be interested in joining in as a maintainer? 😄

**Comment by Cinderella-Man** [comment]
> @Cinderella-Man thanks a lot for the handling of this! Your input to this project has been very helpful. I was wondering if you'd be interested in joining in as a maintainer? smile

Hi @dvcrn, thank you very much for the proposition. I need to admit that I needed to google what's the difference between `Collaborator` and `Maintainer` :slightly_smiling_face:  The short answer is "Yes, I would love to" :+1: The longer answer is - I'm planning to add futures API support but honestly, I don't have much time on my hands those days so it could take a few weeks. I like the fact that this library is really stable for years (partly because of Binance API being backward compatible) and I'm not planning to do any massive changes. There's just the `exchangeInfo` endpoint that changed the result format(that will be a breaking change), but besides that, there's not a lot I could do here - it's great as it is :wink: 

@paulofabiano This PR looks good now - happy to merge, once again,  thank you very much for your contribution :+1:

**Comment by paulofabiano** [comment]
@Cinderella-Man thank you!
I'm new to Elixir but I'm glad I could help the community by implementing it. I can work on something else later.

**Comment by dvcrn** [comment]
@Cinderella-Man fantastic! thanks a lot for the help with this project! :) 

I'll add a step in the CI later that will automatically publish tags to hex so it doesn't need my auth details

**Pull request merged**
